### ISSUE TITLE
Camera shows in Home app, but no snapshot, and can't stream. 

### ISSUE THREAD

**Issue by GreysenPaige** [hjdhjd/homebridge-unifi-protect#848]
**Describe The Problem:**
I have installed the plugin on Homebridge running on my extremely overpowered Synology NAS. (5% of CPU and 1% of RAM being used at most). It shows the camera of my G4 Pro Doorbell in the home app, but it's black. It will not load the snapshots or stream. Logs show tons of red text.

I have installed FFMPEG manually according to this video: https://youtu.be/ai51lXT2zIA?t=124

**To Reproduce:**
Install on Synology NAS, no matter what I do it will not work.

**Logs:**

```
[homebridge.log .txt](https://github.com/hjdhjd/homebridge-unifi-protect/files/10192361/homebridge.log.txt)

```

**Homebridge Configuration:**
synology DS1821+
Version 1.6.0

```json
{
    "controllers": [
        {
            "address": "192.168.86.160",
            "password": "**REMOVED**",
            "username": "**REMOVED**"
        }
    ],
    "platform": "UniFi Protect"
}
```

**Screenshots:**

**Environment:**

* **Homebridge Version:** 1.60
* **Node Version:** v16.16.0
* **Homebridge-UniFi-Protect Plugin Version:** v5.5.4
* **UniFi Protect Version:** 2.6.17
* **UniFi Protect Controller Type:** UniFi Cloud Key Gen2 Plus
* **UniFi Protect Controller Firmware Version:** UCK G2 Plus v3.0.13
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** iOS 16.1.2 iPhone 14 Pro, 16" M1 Max MBP
* **Operating System and OS Version:** Synology NAS DSM 7.1.1-42962 Update 2

**Comment by GreysenPaige** [comment]
Update: I changed the mDNS advertiser to Ciao to fix an issue and now it seems to be working, I'm not sure how the two are related. I'm still showing some red text in the logs. Should I expect to have any issues considering this: 

`[12/9/2022, 1:47:17 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS]: Discovered camera: Front Door [UVC G4 Doorbell Pro] (address: 192.168.86.44 mac: **REMOVED**).
[12/9/2022, 1:47:17 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS]: Controller refresh interval set to 5 seconds.
[12/9/2022, 1:47:17 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: HomeKit Secure Video support enabled.
[12/9/2022, 1:47:17 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Discovered doorbell message switch: LEAVE PACKAGE AT DOOR.
[12/9/2022, 1:47:17 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Discovered doorbell message switch: DO NOT DISTURB.
[12/9/2022, 1:47:27 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS]: Controller API connection terminated because it was taking too long. This error can usually be safely ignored.
[12/9/2022, 1:47:27 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Unable to retrieve a snapshot.
[12/9/2022, 1:48:20 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Streaming request from 192.168.86.153: 1280x720@30fps, 299 kbps. Using 1600x1200@30fps (High), 6000 kbps.
[12/9/2022, 1:48:28 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Stopped video streaming session.
[12/9/2022, 1:48:59 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS]: Controller API connection terminated because it was taking too long. This error can usually be safely ignored.
[12/9/2022, 1:48:59 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Unable to retrieve a snapshot. Using the most recent cached snapshot instead.
[12/9/2022, 1:49:06 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Streaming request from 192.168.86.153: 1280x720@30fps, 299 kbps. Using 1600x1200@30fps (High), 6000 kbps.
[12/9/2022, 1:49:21 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Stopped video streaming session.
`

**Comment by GreysenPaige** [comment]
Continuing to see more issue appearing and showing red in the logs:

> [12/9/2022, 1:54:35 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS]: Controller API connection terminated because it was taking too long. This error can usually be safely ignored.
[12/9/2022, 1:54:35 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Unable to retrieve a snapshot.
[12/9/2022, 1:54:42 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Streaming request from 192.168.86.153: 1280x720@30fps, 299 kbps. Using 1600x1200@30fps (High), 6000 kbps.
[12/9/2022, 1:54:44 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Stopped video streaming session.
[12/9/2022, 1:54:52 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: HomeKit Secure Video event recording enabled: 1600x1200@30fps (High), 2000 kbps with a 4 second timeshift buffer.
[12/9/2022, 1:55:55 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Streaming request from 192.168.86.153: 1280x720@30fps, 299 kbps. Using 1600x1200@30fps (High), 6000 kbps.
[12/9/2022, 1:56:02 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Stopped video streaming session.
[12/9/2022, 1:56:56 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: HomeKit Secure Video has recorded a 24 second motion event.
[12/9/2022, 1:56:56 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: HomeKit Secure Video event recording ended abnormally: The request timed out.
[12/9/2022, 1:56:59 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS]: Controller API connection terminated because it was taking too long. This error can usually be safely ignored.
[12/9/2022, 1:56:59 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Unable to retrieve a snapshot. Using the most recent cached snapshot instead.
[12/9/2022, 1:57:12 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS]: Controller API connection terminated because it was taking too long. This error can usually be safely ignored.
[12/9/2022, 1:57:12 AM] [UniFi Protect] Safety Harbor [UCK-G2-PLUS] Front Door [UVC G4 Doorbell Pro]: Unable to retrieve a snapshot. Using the most recent cached snapshot instead.

**Comment by VelvetMorsels** [comment]
Im having the exact same issues.

**Comment by Parkerbel** [comment]
Hello,
I have a similar issue. The camera G4 PTZ, doesn’t live stream in Apple HomeKit. When I open the Apple Home the image updates along with the other cameras (G3 and G4 Instant). The PTZ doesn’t live stream but the instant cameras work fine.

**Comment by hjdhjd** [comment]
@Parkerbel @VelvetMorsels - one person, one issue please. If you have an issue you would like support on, please open a separate issue and fill out the reporting template, and please don’t hijack another user’s support request with your own, even if you believe you have a similar issue. Thanks for respecting the way I prefer to provide support here.

This thread is for the support request opened by @GreysenPaige.

**Comment by hjdhjd** [comment]
@GreysenPaige Can you provide me a bit of insight on your setup? How many cameras do you have, and what’re you trying to do more broadly?

You mentioned installing your own FFmpeg - what location is it in? In the config you posted, you don’t explicitly point the plugin toward it, which means it’ll search whatever the PATH environment is set to, without necessarily choosing the one you built.

That said, the issue with the controller taking too long to respond is unfortunately a Protect controller issue that we don’t have much influence over. That said, there might be a few things to do there. What’s the quality of the WiFi connection for the doorbell itself? When a snapshot is requested from the controller, the controller will then query the camera. If that’s not happening as quickly as it should, the controller will be especially slow to respond and hang waiting for the request. HomeKit (and therefore HBUP) can’t wait as long as the controller is willing to wait for a response, and you’ll see errors like that.

In the past, when I’ve seen errors like that, as I’ve noted in similar issues before relating specifically to snapshots timing out, it’s come down to either buggy Protect firmwares (and the 2.x series has been notorious for things like this), slow / spotty connectivity of WiFi cameras which cause a further delay to the controller, or overstressed controller platforms (e.g. a dozen+ cameras with high bitrates hitting a UCKGen2+ and all trying to stream at once).

Hope this begins to shed some light for you.

**Comment by GreysenPaige** [comment]
Hey there,

Sorry, I replied via email and it never posted. 

I only have 1 camera at this time. Just the G4 Pro Doorbell. All my other cameras are Nest being connected via a starling hub, so totally separate system and devices with no overlap in resources - and there are only 5 of them and they work great. 

The FFMPEG I installed was just out of desperation in hopes to get something to work, it’s just installed like any other app though. The Wi-Fi connection says “excellent” for the controller and opening the Unifi protect app to see the camera gets me a crystal clear feed in less than 2 seconds.

Thanks for any help you can give!

**Comment by thekeith** [comment]
@GreysenPaige I encountered the same issue. I was able to resolve by setting the option flag "Enable.Video.Stream.Only.Medium" as per this reddit thread: https://www.reddit.com/r/homebridge/comments/txp18c/comment/iaavnq8/

I was receiving the exact same log message with up-to-date firmware on all my Unifi and HomeKit devices. I'm running a Pi 4 with the default installation of ffmpeg from Homebridge as well.
